api/filter_outputs.go

```diff
@@ -136,19 +136,9 @@ func (api *FilterAPI) updateFilterOutput(ctx context.Context, filterOutputID str
 		return err
 	}
 
-	// save the completed event after saving the filter output if its now complete
-	if isNowStatusCompleted {
-		log.Info(ctx, "filter output status is now completed, creating completed event", logData)
-
-		completedEvent := &models.Event{
-			Type: EventFilterOutputCompleted,
-			Time: time.Now(),
-		}
-
-		if err = api.dataStore.AddEventToFilterOutput(ctx, filterOutput.FilterID, completedEvent); err != nil {
-			log.Error(ctx, "failed to add event to filter output", err, logData)
-			return err
-		}
+	if err = api.addEvents(ctx, filterOutput.Events, filterOutputID, isNowStatusCompleted); err != nil {
```

Comment:
Unsure if this change is right. I believe that by calling `addEvents` we will be doing the same as before (adding the completed event) AND also adding all `filterOutput.Events` to the filter output (which wasn't happening previously). Is this intentional? If so, maybe we should reflect it in the tests?

Reply:
It's a good point. I think I was trying to get too fancy where there was no need. I've changed things a bit so there's a helper function for the FilterOutputUpdates to use to make sure we're not adding duplicate events. And then for now I've left the /events POST endpoint as it is - assuming someone intentionally adding a single event will be less likely to inadvertently spam us the same event multiple times.

Added some unit tests for the helper funcs as well